src/controllers/searchController.ts

```diff
@@ -1,14 +1,57 @@
 import { Request, Response, NextFunction } from 'express';
 import { getStallsInclude, fmtStallsResp } from './stallController';
 import { Stall } from '../models';
+import { generatePaginationWithQueries } from '../utils/paginationUtil';
 
 async function searchStalls(req: Request, res: Response, next: NextFunction) {
   try {
     const rawQuery: string | undefined = req.params.query?.trim();
 
+    const categoryFilter = req.query.category;
```

Comment:
If FE gonna call to API using query params with multiple values like this search?category=["Indian Food", "Chinese Food"]. Typecast it to a string should make it work (put req.query.category as string)